File: 7-reverse-integer/reverse-integer.c
int reverse(int x){
    long int res = 0;
        while (x != 0) {
            res *= 10;
            res += x % 10 ;
            x/=  10;
            
        }
        return res > INT32_MAX ? 0 : (res < INT32_MIN ? 0 : res);
}

File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(n) {
    let res = 0

  while (n != 0) {

    res = res * 10 + n % 10

    n = ~~(n / 10)

  }

  return res > 2**31 - 1 ? 0 : res < -1*2**31 ? 0 : res
};